helix-term/src/main.rs

```diff
@@ -84,12 +84,11 @@ FLAGS:
         std::fs::create_dir_all(&conf_dir).ok();
     }
 
-    let config = std::fs::read_to_string(conf_dir.join("config.toml"))
-        .ok()
-        .map(|s| toml::from_str(&s))
-        .transpose()?
-        .or_else(|| Some(Config::default()))
-        .unwrap();
+    let config = match std::fs::read_to_string(conf_dir.join("config.toml")) {
+        Ok(config) => merge_keys(toml::from_str(&config)?),
```

Comment:
And this minor change, it will only merge the keys if we successfully read from a config file.